inc/azure_c_shared_utility/string_token.h

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft. All rights reserved.
```

Comment:
in azure_c_shared_utility, there is alreday string_tokenizer.h/.c. What are the difference between this file and string_tokennizer.h, why cannot we re-use/share code, instead of duplication?

Reply:
If you take a look at https://github.com/Azure/azure-c-shared-utility/tree/master/src you will see they coexist.
Actually this is a newly added string helper by https://github.com/Azure/azure-c-shared-utility/commit/b9a00896f07c8cbab603563ca2660c384ae4c5c5#diff-915d232c49c791133cad712661f76722
As the commitment description mentioned, it's a simplified string tokenizer.

Reply:
ok.